pkg/cookies/csrf.go

```diff
@@ -171,11 +177,11 @@ func decodeCSRFCookie(cookie *http.Cookie, opts *options.Cookie) (*csrf, error)
 // cookieName returns the CSRF cookie's name derived from the base
 // session cookie name
 func (c *csrf) cookieName() string {
-	return csrfCookieName(c.cookieOpts)
+	return csrfCookieName(c.cookieOpts, encryption.HashNonce(c.OAuthState)[0:csrfStateLength - 1])
 }
 
-func csrfCookieName(opts *options.Cookie) string {
-	return fmt.Sprintf("%v_csrf", opts.Name)
+func csrfCookieName(opts *options.Cookie, state string) string {
+	return fmt.Sprintf("%v_%v_csrf", opts.Name, state)
```

Comment:
NIT -- can we potentially flip these so the dynamic part of the cookie name is the suffix? (`%v_csrf_%v`). And it might be worth renaming the `state` variable to signify this is a subset of the state.

Reply:
In the CHANGELOG we will also need to signify this is potentially breaking for any users that had the cookieName option set close to the maximum (255 characters right?) -- Although I hope no one did that 🤣

Reply:
@NickMeves Sure. Fixed both